analyses/tp53_nf1_score/05-evaluate-classifier.py

```diff
@@ -65,45 +65,33 @@
 np.random.seed(123)
 
 # read TP53/NF1 alterations
-status_df = pd.read_table(status_file, low_memory=False)
+full_status_df = pd.read_table(status_file, low_memory=False)
 # read in clinical file
 clinical_df = pd.read_table(clinical)
 # select only IDs
 clinical_df = clinical_df[
     ["Kids_First_Biospecimen_ID", "sample_id", "Kids_First_Participant_ID"]
 ]
 
-
-# add clinical info to alterations dataframe
-status_df = status_df.merge(
-    clinical_df,
-    how="left",
-    left_on="Tumor_Sample_Barcode",
-    right_on="Kids_First_Biospecimen_ID",
-)
-
-# Value count of variant classification
-print(status_df.Variant_Classification.value_counts())
-
-
 # Obtain a binary status matrix
-full_status_df = pd.crosstab(
-    status_df["sample_id"], status_df.Hugo_Symbol, dropna=False
-)
-full_status_df.head(3)
-full_status_df[full_status_df > 1] = 1
-full_status_df = full_status_df.reset_index()
-full_status_df = full_status_df.drop(["No_TP53_NF1_alt"], axis=1)
+full_status_df['tp53_status']= full_status_df.apply(
+    lambda x: (1
+               # IF
+               # both SNV/CNV counts are more than 1 so can be considered as a TP53 double hit
+               if (x['SNV_indel_counts'] !=0 and x['CNV_loss_counts'] != 0 or
+                   # SNV == 1 and cancer_predisposition == Li-Fraumeni syndrome
+                   (x['SNV_indel_counts'] == 1 and (x['cancer_predispositions'] == "Li-Fraumeni syndrome")) or
+                   # SNV == 1 and cancer_predisposition == Li-Fraumeni syndrome
```

Comment:
```suggestion
                   # CNV == 1 and cancer_predisposition == Li-Fraumeni syndrome
```

Reply:
I updated the comment . This is not major but added all the conditions in the for loop chunk since I adding the comment per condition in ifelse was causing a syntax error because of indentation I guess ? 🤔